File: app/src/main/java/com/task/ui/component/details/DetailsActivity.java
package com.task.ui.component.details;

import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;
import com.squareup.picasso.RequestCreator;
import com.task.App;
import com.task.R;
import com.task.data.remote.dto.NewsItem;
import com.task.ui.base.BaseActivity;

import javax.inject.Inject;

import butterknife.BindView;

import static com.task.utils.ObjectUtil.isEmpty;


/**
 * Created by AhmedEltaher on 11/12/16.
 */

public class DetailsActivity extends BaseActivity implements DetailsContract.View {

    @Inject
    DetailsPresenter presenter;

    @BindView(R.id.tv_title)
    TextView tvTitle;
    @BindView(R.id.iv_news_main_Image)
    ImageView ivMainImage;
    @BindView(R.id.tv_description)
    TextView tvDescription;

    @Override
    protected void initializeDagger() {
        App app = (App) getApplicationContext();
        app.getMainComponent().inject(DetailsActivity.this);
    }

    @Override
    protected void initializePresenter() {
        super.presenter = presenter;
        presenter.setView(this);
    }

    @Override
    public int getLayoutId() {
        return R.layout.details_layout;
    }

    @Override
    public void initializeView(NewsItem newsItem) {
        if (!isEmpty(newsItem.getAbstract())) {
            tvDescription.setText(newsItem.getAbstract());
        }
        if (!isEmpty(newsItem.getTitle())) {
            tvTitle.setText(newsItem.getTitle());
        }
        Picasso picasso = Picasso.get();
        RequestCreator requestCreator;
        if (!isEmpty(presenter.getMainImageURL())) {
            requestCreator = picasso.load(presenter.getMainImageURL());
        } else {
            requestCreator = picasso.load(R.drawable.news);
        }
        requestCreator.placeholder(R.drawable.news).into(ivMainImage);
    }
}

File: app/src/main/java/com/task/ui/component/details/DetailsActivity.kt
package com.task.ui.component.details

import android.os.Bundle
import com.squareup.picasso.Picasso
import com.task.R
import com.task.data.remote.dto.NewsItem
import com.task.databinding.DetailsLayoutBinding
import com.task.ui.ViewModelFactory
import com.task.ui.base.BaseActivity
import com.task.utils.Constants
import com.task.utils.observe
import javax.inject.Inject

/**
 * Created by AhmedEltaher
 */

class DetailsActivity : BaseActivity() {

    @Inject
    lateinit var viewModel: DetailsViewModel

    @Inject
    lateinit var viewModelFactory: ViewModelFactory

    private lateinit var binding: DetailsLayoutBinding


    override fun initViewBinding() {
        binding = DetailsLayoutBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = getString(R.string.top_news)
        viewModel.newsItem.value = intent.getParcelableExtra(Constants.NEWS_ITEM_KEY)

    }

    override fun observeViewModel() {
        observe(viewModel.newsItem, ::initializeView)
    }

    override fun initializeViewModel() {
        viewModel = viewModelFactory.create(viewModel::class.java)
    }

    private fun initializeView(newsItem: NewsItem) {
        binding.tvTitle.text = newsItem.title
        binding.tvDescription.text = newsItem.abstractInfo
        if (!newsItem.multimedia.isNullOrEmpty()) {
            Picasso.get().load(newsItem.multimedia[0]?.url).placeholder(R.drawable.news)
                    .into(binding.ivNewsMainImage)
        }
    }
}
